public_html/wp-content/mu-plugins/camptix-tweaks/camptix-tweaks.php

```diff
@@ -919,6 +924,50 @@ function modify_shortcode_contents( $shortcode_contents, $tix_action ) {
 	return $shortcode_contents;
 }
 
+/**
+ * Show error message if IP Address has been throttled by `Form_Spam_Prevention`.
+ */
+function show_throttle_notice() {
+	global $camptix;
+
+	$fsp = new Form_Spam_Prevention(
+		[
+			'prefix' => WC_CAMPTIX_FSP_PREFIX,
+		]
+	);
+
+	if ( $fsp->is_ip_address_throttled() ) {
+		$camptix->error( __( 'You are purchasing tickets too fast. Your IP address has been throttled for an hour since last ticket purchase.', 'wordcamporg' ) );
+
+		// With some payment methods, payment could have been  deducted in the frontend before making a checkout request.
+		// Therefore its important that we disable payment methods tab if we are going to block the checkout request.
+		add_filter( 'tix_render_payment_options', '__return_empty_string', 20 );
+	}
+
+}
+
+/**
+ * Checks if IP is throttled. If not then increments the score by 0.1. This does not handle any sophisticated attack, but is just there so that we do not have to delete junk tickets if a security researcher runs a test on site.
+ *
+ * Maximum score threshold in Form_Spam_Prevention is 4, so using 0.1 implies an IP address will be able to make 39 purchase request before getting throttled.
```

Comment:
`MIGHT` - How did you arrive at `39` as the limit? That feels pretty high to me. I was thinking somewhere closer to `10`; maybe `15` or even `20` if we wanna be real cautious.

I'm open to a higher number, though, but am curious about the reasoning.

Reply:
I was thinking it felt high, too, but then I realized that the way this is implemented, the score increases every time the `form_checkout` callback runs, which I think may be multiple times during a successful ticket purchase.

[MIGHT] Related, since this is a fairly unconventional implementation of `Form_Spam_Prevention` (not using any of the form fields or validation, only the throttle transient), it might be worth adding more explanation to the code comments about how/why it's set up this way so it's easier to grok later on.

Reply:
Oh, that's a good catch. It doesn't seem like we want it to trigger each time, only once per checkout attempt, right?

Reply:
As Corey said, someone might try several times before actually successfully purchasing a ticket, maybe because of payment gateway misbehaving or some other reason. And since this is the ticket purchase page, IMO we should really give benefit of doubt as much as possible.
So my intention is to only block when we are extremely sure that its a bad actor, and even if it is a bot, let it go if its not annoying.

Reply:
👍 

We can always lower it if it becomes a problem in practice.